community/detectors/jira_cve_2022_0540/src/main/java/com/google/tsunami/plugins/detectors/jira/Cve20220540VulnDetector.java

```diff
@@ -0,0 +1,177 @@
+import static com.google.tsunami.common.data.NetworkEndpointUtils.toUriAuthority;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.*;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects the CVE-2022-0540 vulnerability. Reading */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "Cve20220540VulnDetector",
+    version = "0.1",
+    description =
+        "A vulnerability in Jira Seraph allows a remote, unauthenticated attacker to bypass"
+            + " authentication by sending a specially crafted HTTP request. This affects Atlassian"
+            + " Jira Server and Data Center versions before 8.13.18, versions 8.14.0 and later"
+            + " before 8.20.6, and versions 8.21.0 and later before 8.22.0. This also affects"
+            + " Atlassian Jira Service Management Server and Data Center versions before 4.13.18,"
+            + " versions 4.14.0 and later before 4.20.6, and versions 4.21.0 and later before"
+            + " 4.22.0.",
+    author = "thiscodecc",
+    bootstrapModule = Cve20220540DetectorBootstrapModule.class)
+public final class Cve20220540VulnDetector implements VulnDetector {
+
+  @VisibleForTesting static final String INSIGHT_BODY = "Insight Configuration";
+  @VisibleForTesting static final String WBSGANTT_DOBY = "WBS Gantt-Chart";
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String INSIGHT_CHECK_VUL_PATH =
+      "secure/InsightPluginUpdateGeneralConfiguration.jspa;";
+  private static final String WBSGANTT_CHECK_VUL_PATH =
+      "secure/WBSGanttManageScheduleJobAction.jspa;";
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  Cve20220540VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(false).build();
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService);
+  }
+
+  private static String buildTargetInsightUrl(NetworkService networkService) {
```

Comment:
Most of the code for this method is the same as the WBSGantt one. Let's combine both into a single method buildTargetUrl and pass in their unique vuln paths instead.